packages/calcite-components/src/components/shell/shell.tsx

```diff
@@ -193,6 +193,13 @@ export class Shell implements ConditionalSlotComponent {
 
     const contentContainerKey = "content-container";
 
+    const shellPanels: Array<HTMLCalciteShellPanelElement> = Array.from(
+      this.el.querySelectorAll("calcite-shell-panel"),
+    );
+    const hasOnlyBottomPanel: boolean = !!(
+      shellPanels.length === 1 && shellPanels.find((element) => element.slot === "panel-bottom")
```

Comment:
fixme: Instead, use an `onSlotChange` event for the `panel-bottom` slot to determine if a panel bottom shell panel exists and save the state to a State() variable. Also you would need to determine if panel-top slotted element exists and save that to a state. The combined two of those state should give you if one is only panel-bottom. Another state for hasOnlyBottomPanel could be calculated from the two.

Reply:
@driskull I updated the logic to use `onSlotChange` events, please let me know if this looks okay.